sql/moz-fx-data-shared-prod/stripe/itemized_payout_reconciliation/view.sql

```diff
@@ -77,99 +58,72 @@ subscriptions AS (
-    invoice_id
+    invoice_id,
+    country,
+    state,
+    postal_code,
+    currency
 ),
-enriched AS (
+taxes AS (
   SELECT
-    report.* EXCEPT (
-      card_country,
-      shipping_address_city,
-      shipping_address_country,
-      shipping_address_line1,
-      shipping_address_line2,
-      shipping_address_postal_code,
-      shipping_address_state
-    ),
-    {% for country, state, postal_code, dst_pre in [
-        (
-          "customers.shipping_address_country",
-          "customers.shipping_address_state",
-          "customers.shipping_address_postal_code",
-          "shipping_address"
-        ),
-        ("customers.country", "customers.state", "customers.postal_code", "address"),
-        ("card_country", "charge_states.state", "charge_states.postal_code", "card"),
-    ] -%}{# format: off #}
-      CASE
-        -- American Samoa
-        WHEN {{country}} = "US"
-          AND REGEXP_CONTAINS({{postal_code}}, "^96799(-?[0-9]{4})?$")
-          THEN STRUCT("AS" AS `{{dst_pre}}_country`, NULL AS `{{dst_pre}}_state`)
-        -- Puerto Rico
-        WHEN {{country}} = "US"
-          AND REGEXP_CONTAINS({{postal_code}}, "^00[679][0-9]{2}(-?[0-9]{4})?$")
-          THEN STRUCT("PR" AS `{{dst_pre}}_country`, NULL AS `{{dst_pre}}_state`)
-        -- Virgin Islands
-        WHEN {{country}} = "US"
-          AND REGEXP_CONTAINS({{postal_code}}, "^008[0-9]{2}(-?[0-9]{4})?$")
-          THEN STRUCT("VI" AS `{{dst_pre}}_country`, NULL AS `{{dst_pre}}_state`)
-        ELSE STRUCT({{country}} AS `{{dst_pre}}_country`, {{state}} AS `{{dst_pre}}_state`)
-      END.*,
-      {{postal_code}} AS `{{dst_pre}}_postal_code`,
-    {# format: on #}
-    {% endfor -%}
-    subscriptions.* EXCEPT (subscription_id),
+    invoice_id,
+    -- There should only be one set of distinct values for these fields, but if there are more then
+    -- they will be CSVs with the same length and order for consistency
+    STRING_AGG(country, ", " ORDER BY row_number) AS tax_country,
+    STRING_AGG(state, ", " ORDER BY row_number) AS tax_state,
+    STRING_AGG(postal_code, ", " ORDER BY row_number) AS tax_postal_code,
+    STRING_AGG(currency, ", " ORDER BY row_number) AS tax_currency,
+    SUM(tax_amount) AS tax_amount,
   FROM
-    `moz-fx-data-shared-prod`.stripe_external.itemized_payout_reconciliation_v5 AS report
-  LEFT JOIN
-    charge_states
-  USING
-    (charge_id, card_country)
-  LEFT JOIN
-    customers
-  USING
-    (customer_id)
-  LEFT JOIN
-    subscriptions
-  USING
-    (subscription_id)
+    taxes_by_address
+  GROUP BY
+    invoice_id
 )
 SELECT
-  *,
-  -- Use the same address hierarchy as Stripe Tax after we enabled Stripe Tax (FXA-5457).
-  -- https://stripe.com/docs/tax/customer-locations#address-hierarchy
-  -- (customer shipping address, customer billing address, payment method billing address)
+  report.* EXCEPT (card_country),
+  -- This logic preserves legacy behavior for these fields, but these fields are no longer in use
```

Comment:
"These fields" is a little vague, so it might be worth clarifying that you mean the card address fields.